Implemented profile page and updated corresponding components

kevlli opened the pull request:
Implemented the profile page. Including the following changes
- Profile.tsx: the ProfilePage component which is loaded when we go to /profile
- Updated ProfileDropdown.tsx so now it properly routes / links to this Profile page
- Updated App.tsx so Profile page is properly routed to
- Very slight change to FormElementsPage so that we can specify variable width (100%, 75%, etc).

kevlli commented:
I added a DeleteAccount component which is very similar to the LogOut component (basically a popup window that confirms the user actually wants to delete account). Should I refactor and make a parent component with additional props to replace both these components, to avoid code duplication?

kserra1 commented:
This looks like a great start. I do have a couple of recommendations however:

- In this type of page, I don't think you should have to fill out every form to be able to update the page. For ex. you should be able to only update your first name, and put nothing in last name, and press save with no issues.
- Add a range to the date of birth dates. For ex. you shouldn't be able to add your birthday as a date past the currrent date.
- There should be an input field for email to change as well. (Eventually we would have to have some type of verification with the new email)
- There should be an input field for profile picture as well as otherwise user's would never be able to set or change it since we don't allow for that in the sign-up.
- If you have time, the code duplication task would be nice, however not neccesary.

kevlli commented:
Updated with the following functionality:
- email field (no validation yet)
- added validation to date of birth field
- added functionality to not update if the field is empty. no longer requires all fields to submit, only at least 1
- added profile pic update functionality (in-state, backend not integrated yet)
- Also incorporated the UserDTO and UserVM, but I'm not too experienced with that so definitely check it's used correctly

Currently, I believe the workflow for this page is:
- Query some API.  Get the user data.
- Load the data onto the page, mapping to VM first.
- Whenever the user updates some data, we both 
1. Update the data in-state (the Welcome message, email, for ex.)
2. Send this updated data to the backend

kevlli commented:
Abandoned due to rebasing issues. Made a new branch and new pull request - #66
